source/m2mbase.cpp

```diff
@@ -106,10 +106,14 @@ void M2MBase::set_interface_description(const char *desc)
 {
     assert(!_is_static);
     free(_sn_resource->dynamic_resource_params->static_resource_parameters->interface_description_ptr);
-    _sn_resource->dynamic_resource_params->static_resource_parameters->interface_description_ptr =
-            alloc_string_copy((uint8_t*) desc, strlen(desc));
-    _sn_resource->dynamic_resource_params->static_resource_parameters->interface_description_len =
-            strlen(desc);
+    _sn_resource->dynamic_resource_params->static_resource_parameters->interface_description_ptr = NULL;
+    const size_t len = strlen(desc);
+    if (len > 0 ) {
+        _sn_resource->dynamic_resource_params->static_resource_parameters->interface_description_ptr =
+                alloc_string_copy((uint8_t*) desc, strlen(desc));
```

Comment:
You could be reusing the "len" variable instead of "strlen(desc)" to save some code space.